File: tests/cases/classes_scopes.py
def func():
    return "test"

class TestClass:
    def func(self):
        return "test2"

def test():
    num:int = 2
    teststr:str = "test"
    def inner_test():
        return num*teststr
    
    def inner_test_2(num: int):
        return num*teststr
    
    assert inner_test() == "testtest"
    assert inner_test_2(4) == "testtesttesttest"

if __name__ == "__main__":
    assert func() == "test"
    testClass = TestClass()

    assert testClass.func() == "test2"

    # Function scopes
    test()

File: tests/expected/julia/classes_scopes.jl
abstract type AbstractTestClass end
function func()::String
    return "test"
end

mutable struct TestClass <: AbstractTestClass
end
function func(self::TestClass)::String
    return "test2"
end

function test()
    num::Int64 = 2
    teststr::String = "test"
    function inner_test()::String
        return repeat(teststr, num)
    end

    function inner_test_2(num)::String
        return num * teststr
    end

    @assert(inner_test() == "testtest")
    @assert(inner_test_2(4) == "testtesttesttest")
end

if abspath(PROGRAM_FILE) == @__FILE__
    @assert(func() == "test")
    testClass = TestClass()
    @assert(func(testClass) == "test2")
    test()
end
